### PR TITLE
feat(wallet): Make GH Releases for wallet production releases

### LINKED ISSUE #3777: Add github release to wallet production workflow
Draft a github release upon success with:
- Version number in the title `IOTA Wallet vx.y.z`
- Add automatic changelog (filter by wallet relevant commits only)

### PR DESCRIPTION
**Pull request by marc2332**
Closes https://github.com/iotaledger/iota/issues/3777

### CODE REVIEW

File: .github/workflows/apps_wallet_prod_build.yml [hunk 1]
```diff
@@ -39,6 +39,48 @@ jobs:
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ env.CURRENT_TAG }}
+          release_name: IOTA Wallet v${{ steps.version.outputs.version }}
+          body: |
+            ## Changelog
+            {{ env.changelog }}
+          draft: false
+          prerelease: false
```
Comment (DaughterOfMars) [inline .github/workflows/apps_wallet_prod_build.yml]: Are we sure we don't want to have a human review it?

Comment (marc2332) [reply .github/workflows/apps_wallet_prod_build.yml]: The release is merely a formality, it's not related to the actual publishing of the extension or anything like that, you got a fair point though.

@begonaalvarezd wdyt?

Comment (begonaalvarezd) [reply .github/workflows/apps_wallet_prod_build.yml]: lets go for `draft: true`, I prefer revieweing the releases myself

Comment (marc2332) [reply .github/workflows/apps_wallet_prod_build.yml]: we decided to make it `draft` always 🫡

**Review by begonaalvarezd — APPROVED** [review]

File: .github/workflows/apps_wallet_prod_build.yml [hunk 2]
```diff
@@ -39,6 +39,48 @@ jobs:
...
+            changelog=$(cat CHANGELOG.md)
+            echo "changelog=$changelog" >> $GITHUB_ENV
+          fi
+
+      - name: Get version from tag
+        id: version
+        run: echo "::set-output name=version::${GITHUB_REF#refs/tags/wallet-v}"
+
+      - name: Create GitHub Release
+        uses: elgohr/create-release@v5
```
Comment (thibault-martinez) [inline .github/workflows/apps_wallet_prod_build.yml]: please pin to commit

File: .github/workflows/apps_wallet_prod_build.yml [hunk 3]
```diff
@@ -39,6 +39,48 @@ jobs:
...
+      - name: Generate Changelog
+        id: generate_changelog
+        run: |
+          if [ "${{ env.PREV_TAG }}" = "none" ]; then
+            echo "No previous tag found. Skipping changelog generation."
+            echo "changelog=No previous tag found. Changelog generation skipped." >> $GITHUB_ENV
+          else
+            git log ${{ env.PREV_TAG }}..${{ env.CURRENT_TAG }} --pretty=format:"- %s by @%an in #%h" -- ./apps/wallet > CHANGELOG.md
+            changelog=$(cat CHANGELOG.md)
+            echo "changelog=$changelog" >> $GITHUB_ENV
```
Comment (thibault-martinez) [inline .github/workflows/apps_wallet_prod_build.yml:65]: Out of curiosity, when is GITHUB_ENV used then?

Comment (marc2332) [reply .github/workflows/apps_wallet_prod_build.yml:65]: `$GITHUB_ENV` is a not a normal env, its how you can move a locally-defined env var to the entire workflow so the next steps can use what you put in there, kinda like outputs

In this case, I am putting the `$changelog` env var

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable

Comment (thibault-martinez) [reply .github/workflows/apps_wallet_prod_build.yml:65]: Thanks!

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**